### PR TITLE
How to create and run a .NET Unit Testing Project

### LINKED ISSUE #3596: How to create and run a .NET Unit Testing Project
### Proposed title of the article
How to create and run a .NET Unit Testing Project

### Introduction paragraph (2-3 paragraphs):
Testing is an essential aspect of any programming language. Testing for ASP.Net applications is possible with the help of Visual Studio.
Visual Studio is used to create test code. It is also used to run the test code for an ASP.Net application. In this way, it becomes simple to check for any errors in an ASP.Net application. In Visual Studio, the testing module comes with an out of box functionality. One can straightaway perform a test for an ASP.Net project.
The first level of testing an ASP.Net project is unit level testing. This test is the functionality of an application. The testing is conducted to ensure that the application behaves as expected. In ASP.Net, the first task is to create a test project in Visual Studio. The test project will contain the necessary code to test the application.

### Key takeaways: 
By the end of this tutorial the learner will understand:
- What is unit testing for ASP.Net?
- Steps and how to implement testing
- How to create a .NET Unit Testing Project
- How to run the test using Visual Studio

### References: 
N/A

### PR DESCRIPTION
**Pull request by Ushter21**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #3596 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+### Creating a project
+
+1. When creating a new project, we will use Visual Studio and create a new project from the start menu.
+2. Once we've done that, we'll need to locate our.NET Core's C# console app project template and click the icon that appears there.
+3. Decide on a name for your endeavor and then get to operate. Either the.NET Core 3.1 or.NET 5 target framework will have to be selected.
+
+   As you work on the School project, the Program.cs file will be open in the code editor.
+
+   > If you are unable to access the Program.cs file directly, you may do so by double-clicking it in Solution Explorer.
```
Comment (zolomohan) [inline content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md:73]: Please add screenshots for visual guidance.

File: content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+We have a project that we can now be put to the test.
+
+### How to create a unit-testing project
+
+1. Add a new project simply from File Menu.
+2. You have to choose the language of choice, which in our case is C#, then proceed after you have selected the unit test project for the .NET Core template.
+3. Rename such as CollegeTests, then proceed by clicking next.
+4. Create a new CollegeTests project with the desired architecture (.NET Core 3.1 or.NET 5) selected. We've integrated CollegeTests into the School solution.
+5. Make sure the CollegeTests project contains a mention of the College project. In Solution Explorer, go to CollegeTests and add a dependency by clicking Dependencies under CollegeTests.
+6. When you open the Reference Manager dialog box, expand Projects, select Solution, and select College from the Reference Manager drop-down menu. Click OK to continue.
```
Comment (zolomohan) [inline content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md:140]: Please add screenshots for visual guidance.

File: content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+- It gives a void result.
+- It isn't allowed to have parameters.
+
+### Create the test and execute it
+
+1. Choose Build Solution from the Build drop-down menu.
+2. If Test Explorer isn't already open, go to test > Windows, then select Test Explorer from the main menu.
+3. Run the test from the Run All section. While running the test, a progress bar appears and changes color. In the event that all tests are successful, then the bar is green; otherwise, it is red.
+The test will fail if we assume our worst-case scenario.
+4. To explore more about a technique, open Test Explorer and click on the technique's name.
```
Comment (zolomohan) [inline content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md:217]: Please add screenshots for visual guidance.

**Review by zolomohan — CHANGES_REQUESTED** [review]

**Comment by zolomohan** [comment]
Hi @Ushter21. Please add screenshots for visual guidance. Ping me when done.

File: content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+4. To use the 'CollegeFinance' class, add the following C# code to Program.cs:
+
+```C#
+using System;
+namespace CollegeFinanceNS
+{   
+  public class CollegeFinance
+  {
+    private readonly string n_studentName;
```
Comment (zolomohan) [inline content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md:83]: The variable name is `n_studentName` here but `n_studentIdentity` is being used below.

File: content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+      n_admNo += total; // purposefully wrong code
+    }
+    public void Fees (double total)
+    {
+      if (total < 0.0)
+      {
+        throw new ArgumentOutOfRangeException("fees";
+      }
+      n_admNo += total;
+    }
```
Comment (zolomohan) [inline content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md:118]: Duplicate code? These functions have the same name with the same signature. The second function has one if condition missing.

File: content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+    public static void Main()
+    {
+      CollegeFinance ba = new CollegeFinance("Anthony Lincons", 3235);
+      ba.Balance(754.50);
```
Comment (zolomohan) [inline content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md:122]: Balance is not definend.

File: content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+      ba.Fees(2345.00);
+      Console.WriteLine("Current Balance is ${0}", ba.Balance;
```
Comment (zolomohan) [inline content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md:124]: There is no get for ba.Balance?

File: content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+```C#
+[TestMethod]
+public void Fees_WithValidTotal_UpdatesAdmNo()
+{ 
+  double beginningAdmNo = 277389;
+  double FeesTotal = 3257;
+  double expected = 1245;
+  CollegeFinance finance = new CollegeFinance("Antony Lincons", beginningAdmNo);
+  finance.Fees(feesTotal);
+  double actual = finance.AdmNo;
```
Comment (zolomohan) [inline content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md:196]: The get is set as `admNo` but here it's `AdmNo`

File: content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+  Assert.AreEqual(expected, actual, 2334, "Fees not summed correctly");
```
Comment (zolomohan) [inline content/articles/how-to-create-and-run-a-.net-unit-testing -project/index.md:197]: What is the 3rd argument `2334`?

**Review by zolomohan — CHANGES_REQUESTED** [review]

**Comment by hectorkambow** [comment]
@Ushter21 
Good afternoon and thank you for submitting your topic to the EngEd program. It seems this PR is still quite a bit aways from being completed. 
As a rule of thumb - please only submit articles (pull requests) with a **complete and fully polished article - ready to be published.** 
All revisions and edits should be completed on your own forked repo (as to not build up a backlog in the queue). 

We will be closing this and please re-submit when it is ready to be published as a new PR (and you can add the duplicate label) and ping the reviewer and myself. 


cc @zolomohan

**Pull request closed without merging**